appinventor/components-ios/src/Barometer.swift

```diff
@@ -0,0 +1,88 @@
+// -*- mode: swift; swift-mode:basic-offset: 2; -*-
+// Copyright 2023 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+import Foundation
+import CoreMotion
+
+class Barometer: NonvisibleComponent {
+  private let motionManager = CMAltimeter()
+  private var refreshTimer: Timer?
+  private var enabled = true
+  @objc var AirPressure: NSNumber = 0.0
+  @objc var Available: Bool = false
+  @objc var RefreshTime: TimeInterval = 0.001
+  
+  override init(_ container: ComponentContainer) {
+    super.init(container)
+    checkAvailability()
+    startListening()
```

Comment:
Rather than starting the listener here, implement `@objc open func Initialize()` so that the listening can start during the initialization phase, and only if `enabled`. Otherwise, this may prompt the user for permission even if Enabled = false is set by the app developer at the start of the app.